p4-16/psa/PSA.mdk

```diff
@@ -1161,11 +1161,26 @@ corresponding table entries.
 [INCLUDE=psa.p4:ActionSelector_extern]
 ```
 
-## Packet Generation
+## Digest
+
+A PSA implementation provides a digest mechanism to send a separate
+data object to the control plane. The data object may contain a
+identifier to identify the receiver of the object. The data object can
+optionally attach metadata to provide more information about the
+source of the digest. The attached metadata can be one or more header
+fields from a packet or one or more metadata generated in the Ingress
+or Egress pipeline. The digest extern can only be invoked in the
+deparser.
+
+A common use case of the digest mechanism is MAC learning. In the MAC
+learning case, the receiver of the digest is the CPU port, represented
+by a enum type. The digest metadata is the source MAC address of the
+received packet.
 
-\TODO: is generating a new packet and sending it to the stream or is
-it adding a header to the current packet and sending it to the
-stream (copying or redirecting).
+It may be desirable to avoid sending a burst of identical digests to
+the control plane, to save the processing cycles of the control plane
+processor for other tasks. In this case, the PSA implementation can
+implement a deduplication mechanism to filter repeated digests. 
```

Comment:
may implement. Since we're not providing any guarantees on latency and any support for aggregation, is it even worth talking about this? Maybe we can move this to a "note to implementers".